engine/src/main/java/io/zeebe/engine/state/appliers/ProcessInstanceElementActivatingApplier.java

```diff
@@ -30,6 +31,17 @@ public void applyState(final long elementInstanceKey, final ProcessInstanceRecor
     elementInstanceState.newInstance(
         flowScopeInstance, elementInstanceKey, value, ProcessInstanceIntent.ELEMENT_ACTIVATING);
 
+    final var flowScopeElementType = flowScopeInstance.getValue().getBpmnElementType();
+    final var currentElementType = value.getBpmnElementType();
+    if ((currentElementType == BpmnElementType.START_EVENT
+            && (flowScopeElementType == BpmnElementType.SUB_PROCESS
+                || flowScopeElementType == BpmnElementType.PROCESS))
+        || flowScopeElementType == BpmnElementType.MULTI_INSTANCE_BODY) {
```

Comment:
Can we extract this to a helper? e.g. `isFirstElementInContainerScope()` or something with a better name. I definitely had to read this very carefully to understand what the condition is.

Reply:
I extract the conditional expression